### PR TITLE
Implement try catch in Components > Wz files

### LINKED ISSUE #3447: Refactor try catch in Wz files in Components
| Parent Issue |
|:---:|
|#3265| 

**Description**
In order to improve better error handling in the front-end, we should implement a better try-catch strategy in wz files in `Components` section, services, controller.

### PR DESCRIPTION
**Pull request by pablomarga**
Hi team,
This PR implement our new error handling strategy on the wz files in components section

Close #3447

### CODE REVIEW

File: public/components/wz-search-bar/lib/suggest-handler.ts [hunk 1]
```diff
@@ -108,7 +111,17 @@ export class SuggestHandler extends BaseHandler {
...
+        context: `${SuggestHandler.name}.checkQuery`,
+        level: UI_LOGGER_LEVELS.ERROR,
+        severity: UI_ERROR_SEVERITIES.BUSINESS,
+        error: {
+          error: error,
+          message: error.message || error,
+          title: error.name || error,
+        },
+      };
+      getErrorOrchestrator().handleError(options);
```
Comment (Desvelao) [inline public/components/wz-search-bar/lib/suggest-handler.ts:124]: **thought**: Before the changes there was a `return {error}` and after that returns is removed, this changes the logic, and maybe that returns could be expected and managed in other places. We should check before removing it.

File: public/components/wz-filter-bar/wz-filter-bar.js [hunk 2]
```diff
@@ -14,6 +14,9 @@ import { EuiComboBox } from '@elastic/eui';
 import PropTypes from 'prop-types';
 import './wz-filter-bar.scss';
 import { withErrorBoundary } from '../common/hocs';
+import { UI_LOGGER_LEVELS } from '../../../common/constants';
+import { UI_ERROR_SEVERITIES } from '../../react-services/error-orchestrator/types';
+import { getErrorOrchestrator } from '../../react-services/common-services'
```
Comment (Desvelao) [inline public/components/wz-filter-bar/wz-filter-bar.js:19]: **nitpick**: Add the semicolon `;`.

File: public/components/wz-filter-bar/wz-filter-bar.js [hunk 3]
```diff
@@ -14,292 +14,302 @@ import { EuiComboBox } from '@elastic/eui';
...
+          q: queryObj.query,
+          search: queryObj.search,
+          selectedOptions: selectedOptions,
+        });
+      } catch (error) {
+        const options = {
+          context: `${WzFilterBar.name}.buildQuery`,
+          level: UI_LOGGER_LEVELS.ERROR,
+          severity: UI_ERROR_SEVERITIES.BUSINESS,
+          store: true,
```
Comment (frankeros) [inline public/components/wz-filter-bar/wz-filter-bar.js:281]: always when a query build fails will it be stored? can it be caused by a user intentionally?

File: public/components/wz-search-bar/wz-search-bar.tsx [hunk 4]
```diff
@@ -11,90 +11,133 @@
...
+          if (error !== 'New request in progress') {
+            const options = {
+              context: `${useSuggestHandler.name}.useEffect`,
+              level: UI_LOGGER_LEVELS.ERROR,
+              severity: UI_ERROR_SEVERITIES.CRITICAL,
+              store: true,
+              error: {
+                error: error,
+                message: error.message || error,
+                title: error.name || error,
```
Comment (frankeros) [inline public/components/wz-search-bar/wz-search-bar.tsx:128]: what happen if error is an object?

**Review by gabiwassan — APPROVED** [review]
LGTM

**Pull request merged**